testing/board/test_board_dc.cpp

```diff
@@ -0,0 +1,319 @@
+	if (fd < 0) {
+		printf("open failed \n");
+		return FPGA_NOT_FOUND;
+	}
+
+	ssize_t total_written = eintr_write(fd, buf, count);
+	if (total_written == 0) {
+		close(fd);
+		printf("total_written failed \n");
+		return FPGA_INVALID_PARAM;
+	}
+
+	close(fd);
+	return res;
+}
+
+fpga_result board_dc_c_p::delete_sysfs_file(const char *file) {
+	fpga_result res = FPGA_OK;
+	char sysfspath[FPGA_VAR_BUF_LEN];
+	int status = 0;
+
+	snprintf_s_ss(sysfspath, sizeof(sysfspath), "%s/%s", "/sys/class/fpga/intel-fpga-dev.0/intel-fpga-fme.0", file);
+
+	glob_t pglob;
+	int gres = glob(sysfspath, GLOB_NOSORT, NULL, &pglob);
+	if ((gres) || (1 != pglob.gl_pathc)) {
+		globfree(&pglob);
+		return FPGA_NOT_FOUND;
+	}
+	status = remove(pglob.gl_pathv[0]);
+
+	globfree(&pglob);
+	if (status < 0) {
+		printf("delete failed = %d \n", status);
+		return FPGA_NOT_FOUND;
+	}
+
+	return res;
+}
+
+/**
+* @test       board_dc_1
+* @brief      Tests: read_bmcfw_version
+* @details    Validates bmc firmware version  <br>
+*/
+TEST_P(board_dc_c_p, board_dc_1) {
+
+	char bmcfw_ver[FPGA_VAR_BUF_LEN];
+
+	EXPECT_EQ(read_bmcfw_version(tokens_[0], bmcfw_ver, FPGA_VAR_BUF_LEN), FPGA_OK);
+
+	EXPECT_EQ(read_bmcfw_version(tokens_[0], NULL, FPGA_VAR_BUF_LEN), FPGA_INVALID_PARAM);
+
+	EXPECT_EQ(read_bmcfw_version(NULL, bmcfw_ver, FPGA_VAR_BUF_LEN), FPGA_INVALID_PARAM);
+}
+
+/**
+* @test       board_dc_2
+* @brief      Tests: read_max10fw_version
+* @details    Validates max10 firmware version  <br>
+*/
+TEST_P(board_dc_c_p, board_dc_2) {
+
+	char max10fw_ver[FPGA_VAR_BUF_LEN];
+
+	EXPECT_EQ(read_max10fw_version(tokens_[0], max10fw_ver, FPGA_VAR_BUF_LEN), FPGA_OK);
+
+	EXPECT_EQ(read_max10fw_version(tokens_[0], NULL, FPGA_VAR_BUF_LEN), FPGA_INVALID_PARAM);
+
+	EXPECT_EQ(read_max10fw_version(NULL, max10fw_ver, FPGA_VAR_BUF_LEN), FPGA_INVALID_PARAM);
+}
+
+
+/**
+* @test       board_dc_4
+* @brief      Tests: print_board_info
+* @details    Validates fpga board info  <br>
+*/
+TEST_P(board_dc_c_p, board_dc_4) {
+
+	EXPECT_EQ(print_board_info(tokens_[0]), FPGA_OK);
+	EXPECT_NE(print_board_info(NULL), FPGA_OK);
+}
+
+/**
+* @test       board_dc_5
+* @brief      Tests: parse_fw_ver
+* @details    Validates parse fw version  <br>
+*/
+TEST_P(board_dc_c_p, board_dc_5) {
+
+	char buf[10] = {0};
+	char buf_ver[10] = {0};
+
+	EXPECT_EQ(parse_fw_ver(buf,buf_ver,strlen(buf)), FPGA_OK);
+
+	EXPECT_EQ(parse_fw_ver(NULL, buf_ver, strlen(buf)), FPGA_INVALID_PARAM);
+	EXPECT_EQ(parse_fw_ver(buf, NULL, strlen(buf)), FPGA_INVALID_PARAM);
+
+	EXPECT_EQ(parse_fw_ver(buf, buf_ver, 0x100000), FPGA_OK);
+}
+
+/**
+* @test       board_dc_6
+* @brief      Tests: read_mac_info
+* @details    Validates read mac info  <br>
+*/
+TEST_P(board_dc_c_p, board_dc_6) {
+
+	struct ether_addr mac_addr;
+
+	EXPECT_EQ(read_mac_info(tokens_[0], 0, &mac_addr), FPGA_OK);
+	EXPECT_EQ(read_mac_info(tokens_[0], 0, NULL), FPGA_INVALID_PARAM);
+
+	EXPECT_EQ(read_mac_info(tokens_[0], 200, &mac_addr), FPGA_INVALID_PARAM);
```

Comment:
why does this return FPGA_INVALID_PARAM?

Reply:
fixed

Reply:
Invalid AFU channel